src/mjolnir/shortcutbuilder.cc

```diff
@@ -472,6 +483,12 @@ uint32_t AddShortcutEdges(GraphReader& reader,
           break;
         }
 
+        //terminate shortcut if edge length has exceeded
+        shortcutLength +=tile->directededge(next_edge_id)->length(); 
+        if(shortcutLength > kMaxEdgeLength){
+          break; 
+        }
```

Comment:
you can remove the return from ConnectEdges() and do this instead:

```suggestion
        length += tile->directededge(next_edge_id)->length(); 
        if(length > kMaxEdgeLength){
          break; 
        }
        
        // further down
        ConnectEdges(..)
```